Libraries/Modal/Modal.js

```diff
@@ -23,29 +23,94 @@ const requireNativeComponent = require('requireNativeComponent');
 const RCTModalHostView = requireNativeComponent('RCTModalHostView', null);
 
 /**
- * A Modal component covers the native view (e.g. UIViewController, Activity)
- * that contains the React Native root.
+ * The Modal component is a simple way to present content above an enclosing view.
  *
- * Use Modal in hybrid apps that embed React Native; Modal allows the portion of
- * your app written in React Native to present content above the enclosing
- * native view hierarchy.
+ * _Note: If you need more control over how to present modals over the rest of your app,
+ * then consider using a top-level Navigator. Go [here](/react-native/docs/navigator-comparison.html) to compare navigation options._
  *
- * In apps written with React Native from the root view down, you should use
- * Navigator instead of Modal. With a top-level Navigator, you have more control
- * over how to present the modal scene over the rest of your app by using the
- * configureScene property.
+ * 
+ * ```javascript
+ * var React = require('react');
+ * var ReactNative = require('react-native');
+ * var {
+ *   Modal, Text, TouchableHighlight, View, AppRegistry,
+ * } = ReactNative;
+ * var ModalExample = React.createClass({
+ *   getInitialState() {
+ *     return {
+ *       modalVisible: false,
+ *     };
+ *   },
+ *   setModalVisible(visible) {
+ *     this.setState({modalVisible: visible});
+ *   },
+ *   render() {
+ *     return (
+ *       <View style={{marginTop: 22}}>
+ *         <Modal
+ *           animationType={"slide"}
+ *           transparent={false}
+ *           visible={this.state.modalVisible}
+ *           onRequestClose={() => {alert("Modal has been closed.")}}
+ *           >
+ *          <View style={{marginTop: 22}}>
+ *           <View>
+ *             <Text>Hello World!</Text>
+ * 
+ *             <TouchableHighlight onPress={() => {
+ *               this.setModalVisible(!this.state.modalVisible)
+ *             }}>
+ *               <Text>Hide Modal</Text>
+ *             </TouchableHighlight>
+ * 
+ *           </View>
+ *          </View>
+ *         </Modal>
+ *         <TouchableHighlight onPress={() => {
+ *           this.setModalVisible(true)
+ *         }}>
+ *           <Text>Show Modal</Text>
+ *         </TouchableHighlight>
+ *       </View>
+ *     );
+ *   },
+ * });
+ * 
+ * // App registration and rendering
+ * AppRegistry.registerComponent('AwesomeProject', () => ModalExample);* ```
  */
```

Comment:
Let's make the example ES6 for consistency with the other changes we're making and also drop the AppRegistry part.